specs/GetFavicon.md

```diff
@@ -0,0 +1,134 @@
+# Background
+A Favicon is an asset which is a part of a  webpage, and typically displayed on each tab. Developers would
+like to have an API which allows them to retrieve the Favicon of a webpage, if it has one, as well as get an update whenever
+the favicon has changed.
+
+# Description
+We propose a new Webview2 event which would allow developers access the current Favicon of a page, 
+as well as be notified when the favicon changes.
+
+# Examples
+## Win32 C++ Registering a listener for favicon changes
+```cpp
+    CHECK_FAILURE(m_webView->add_FaviconChanged(
+        Callback<ICoreWebView2FaviconChangedEventHandler>(
+            [this](ICoreWebView2* sender, IUnknown* args) -> HRESULT {
+                
+                wil::unique_cotaskmem_string url;
+                CHECK_FAILURE(sender->get_FaviconUri(&url));
+                wil::com_ptr<IStream> pStream = SHCreateMemStream(nullptr, 0);
+
+                sender->GetFavicon(
+                    COREWEBVIEW2_FAVICON_IMAGE_FORMAT_PNG, 
+                    pStream,
+                    Callback<
+                        ICoreWebView2GetFaviconCompletedHandler>(
+                        [&pStream, this](HRESULT code) -> HRESULT {
+                                    Gdiplus::Bitmap* pBitmap = new Gdiplus::Bitmap(pStream);
+                                    HICON icon;
+                                    pBitmap->GetHICON(&icon);
+                                    SendMessage(
+                                        m_appWindow->GetMainWindow(), WM_SETICON, ICON_SMALL,
+                                        (LPARAM)icon);
+                            return S_OK;
+                        })
+                        .Get());
+
+                return S_OK;
+            })
+            .Get(),
+        &m_faviconChangedToken));  
+```
+## .NET / WinRT Registering a listener for favicon changes
+```c#
+webView.CoreWebView2.FaviconChanged += (CoreWebView2 sender, Object arg) =>
+{
+    System.IO.Stream stream = new System.IO.MemoryStream();
+    await webView.CoreWebView2.GetFaviconAsync(
+        CoreWebView2FaviconImageFormat.Png,
+        stream);
+    // setting the window Icon to the bitmap
+    this.Icon = BitmapFrame.Create(stream); 
+
+};
+```
+# API Notes
+If a Web page does not have a Favicon, then the event is not fired.
```

Comment:
This statement is a little confusing. The event is raised when the favicon changes. Presumably if it changes from us having a favicon to not having a favicon the event will be raised? If a page changes its favicon from a favicon to no favicon or we navigate from a page with a favicon to a page without a favicon then there must be some way for the end dev to know there's no longer a favicon. I would imagine we should raise the event in this case?

Reply:
As a note also answer - Also what happens if we call GetFaviconAsync and there is no favicon?
here

Reply:
Once I determine these details, I will add in more comments too (per Liangs request).